db/db_bloom_filter_test.cc

```diff
@@ -2618,8 +2618,7 @@ TEST_F(DBBloomFilterTest, OptimizeFiltersForHits) {
       BottommostLevelCompaction::kSkip;
   compact_options.change_level = true;
   compact_options.target_level = 7;
-  ASSERT_TRUE(db_->CompactRange(compact_options, handles_[1], nullptr, nullptr)
-                  .IsNotSupported());
+  ASSERT_OK(db_->CompactRange(compact_options, handles_[1], nullptr, nullptr));
```

Comment:
[Note] Due to some clean up in RefitLevel, this now returns OK. The result of this CompactRange() does not matter for this UT after inspection.

Reply:
> It's due to the new conditional check where level is 8

Correct.

> It seems that we may incorrectly compute final_output_level

Thanks! This is an accidental finding which I plan to address it in https://github.com/facebook/rocksdb/pull/11041. The test has been using the wrong final_output_level prior my PR anyway.